rcldotnet/RCLdotnet.cs

```diff
@@ -198,9 +221,36 @@ public class RCLdotnet {
 
     public static Node CreateNode (string nodeName, string nodeNamespace) {
       IntPtr nodeHandle = IntPtr.Zero;
-      int ret = RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
-      Node node = new Node (nodeHandle);
-      return node;
+      RMWRet ret = (RMWRet)RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
+
+      switch (ret) {
+        case RMWRet.NodeNameValid:
+          Node node = new Node (nodeHandle);
+          return node;
+        case RMWRet.NodeNameInvalidIsEmptyString:
+          throw new NodeNameInvalidIsEmptyStringException(RMWGetErrorString());
+        case RMWRet.NodeNameInvalidContainsUnallowedCharacters:
+          throw new NodeNameInvalidContainsUnallowedCharactersException(RMWGetErrorString());
+        case RMWRet.NodeNameInvalidStartsWithNumber:
+          throw new NodeNameInvalidStartsWithNumberException(RMWGetErrorString());
+        case RMWRet.NodeNameInvalidTooLong:
+          throw new NodeNameInvalidTooLongException(RMWGetErrorString());
+        default:
+          switch((RCLRet)ret) {
+              case RCLRet.AlreadyInit:
+                throw new NodeNameInvalidTooLongException(RCLGetErrorString());
+              case RCLRet.InvalidArgument:
+                throw new InvalidArgumentException(RCLGetErrorString());
+              case RCLRet.BadAlloc:
+                throw new BadAllocException(RCLGetErrorString());
+              case RCLRet.NodeInvalidName:
+                throw new NodeInvalidNameException(RCLGetErrorString());
+              case RCLRet.NodeInvalidNamespace:
+                throw new NodeInvalidNamespaceException(RCLGetErrorString());
+              default:
+                return null;
```

Comment:
If the return code does not match any of the know ones, this should raise a system exception (like `SystemException`) with an appropriate description.

Reply:
I throw an InvalidOperationException. I think I cannot throw a SystemException directly

Reply:
SystemException works for me:

https://gist.github.com/esteve/107af25d607f93ca12807c0ffdb0f994

```
esteve@eobard:/tmp/d/foo$ dotnet run
Hello World!

Unhandled Exception: System.SystemException: Unknown error
   at foo.Program.Main(String[] args) in /tmp/d/foo/Program.cs:line 10
```

Reply:
But in any case, InvalidOperation is fine.